enginetest/histogram_test.go

```diff
@@ -0,0 +1,514 @@
+				mean := args[0].(float64)
+				std := args[1].(float64)
+				xyz := makeTable(db, fmt.Sprintf("xyz%d", tab), tab, col)
+				err := normalDistForTable(ctx, xyz, cnt, mean, std, int(tab))
+				if err != nil {
+					panic(err)
+				}
+				return xyz
+			},
+			leftOrd:    []int{1},
+			rightOrd:   []int{1},
+			leftTypes:  []sql.Type{types.Int64, types.Int64, types.Int64},
+			rightTypes: []sql.Type{types.Int64, types.Int64, types.Int64},
+			err:        1,
+			args1:      []interface{}{t.mean1, t.std1},
+			args2:      []interface{}{t.mean2, t.std2},
+		}
+		statTests = append(statTests, st)
+	}
+
+	debug := false
+	runSuite(t, statTests, 100, 5, debug)
+	runSuite(t, statTests, 100, 10, debug)
+	runSuite(t, statTests, 100, 20, debug)
+	runSuite(t, statTests, 500, 10, debug)
+	runSuite(t, statTests, 500, 20, debug)
+}
+
+func TestExpDist(t *testing.T) {
+	tests := []struct {
+		name    string
+		lambda1 float64
+		lambda2 float64
+	}{
+		{
+			name:    "same table",
+			lambda1: 1,
+			lambda2: 1,
+		},
+		{
+			name:    ".5/1.5",
+			lambda1: .5,
+			lambda2: 1.5,
+		},
+		{
+			name:    ".25/3",
+			lambda1: .25,
+			lambda2: 3,
+		},
+	}
+
+	var statTests []statsTest
+	for _, tt := range tests {
+		st := statsTest{
+			name: tt.name,
+			tableGen: func(ctx *sql.Context, db *memory.Database, cnt int, tab sql.TableId, col sql.ColumnId, args ...interface{}) *plan.ResolvedTable {
+				xyz := makeTable(db, "xyz", tab, col)
+				err := expDistForTable(ctx, xyz, cnt, args[0].(float64), int(tab))
+				if err != nil {
+					panic(err)
+				}
+				return xyz
+			},
+			leftOrd:    []int{1},
+			rightOrd:   []int{1},
+			leftTypes:  []sql.Type{types.Int64, types.Int64, types.Int64},
+			rightTypes: []sql.Type{types.Int64, types.Int64, types.Int64},
+			args1:      []interface{}{tt.lambda1},
+			args2:      []interface{}{tt.lambda2},
+			err:        1,
+		}
+		statTests = append(statTests, st)
+	}
+
+	debug := false
+	runSuite(t, statTests, 100, 5, debug)
+	runSuite(t, statTests, 100, 10, debug)
+	runSuite(t, statTests, 100, 20, debug)
+	runSuite(t, statTests, 500, 10, debug)
+	runSuite(t, statTests, 500, 20, debug)
+}
+
+func TestMultiDist(t *testing.T) {
+	tests := []statsTest{
+		{
+			name: "uniform dist int",
+			tableGen: func(ctx *sql.Context, db *memory.Database, cnt int, tab sql.TableId, col sql.ColumnId, args ...interface{}) *plan.ResolvedTable {
+				xyz := makeTable(db, "xyz", tab, col)
+				err := uniformDistForTable(ctx, xyz, cnt)
+				if err != nil {
+					panic(err)
+				}
+				return xyz
+			},
+			leftOrd:    []int{1},
+			rightOrd:   []int{1},
+			leftTypes:  []sql.Type{types.Int64, types.Int64, types.Int64},
+			rightTypes: []sql.Type{types.Int64, types.Int64, types.Int64},
+			err:        .1,
+		},
+	}
+
+	runSuite(t, tests, 1000, 10, false)
+}
+
+func runSuite(t *testing.T, tests []statsTest, rowCnt, bucketCnt int, debug bool) {
```

Comment:
Can you add a docstring explaining the what `runSuite` is evaluating. I gather that it's getting the histogram of the join between the tables, but what is it asserting about the histogram?